node/core/core.go

```diff
@@ -11,44 +13,31 @@ import (
 
 // Config combines all configuration fields for managing the relationship with a Core node.
 type Config struct {
-	Remote       bool
-	RemoteConfig struct {
-		Protocol   string
-		RemoteAddr string
-	}
+	Protocol   string
+	RemoteAddr string
 }
 
 // DefaultConfig returns default configuration for Core subsystem.
 func DefaultConfig() Config {
-	return Config{
-		Remote: false,
-	}
+	return Config{}
 }
 
 // Components collects all the components and services related to managing the relationship with the Core node.
-func Components(cfg Config, loader core.RepoLoader) fxutil.Option {
+func Components(cfg Config) fxutil.Option {
 	return fxutil.Options(
 		fxutil.Provide(core.NewBlockFetcher),
 		fxutil.ProvideAs(header.NewCoreExchange, new(header.Exchange)),
 		fxutil.Invoke(HeaderCoreListener),
-		fxutil.ProvideIf(cfg.Remote, func() (core.Client, error) {
-			return RemoteClient(cfg)
-		}),
-		fxutil.InvokeIf(cfg.Remote, func(c core.Client) error {
-			return c.Start()
-		}),
-		fxutil.ProvideIf(!cfg.Remote, func() (core.Client, error) {
-			store, err := loader()
-			if err != nil {
-				return nil, err
+		fxutil.Provide(func() (core.Client, error) {
+			if cfg.RemoteAddr == "" {
+				return nil, fmt.Errorf("no celestia-core endpoint given")
 			}
-
-			cfg, err := store.Config()
+			client, err := RemoteClient(cfg)
 			if err != nil {
 				return nil, err
 			}
-
-			return core.NewEmbedded(cfg)
+			err = client.Start()
```

Comment:
Wrap it with hook

Reply:
`client.Start` and `client.Stop` do not take context :(

Reply:
jk, resolved.

Reply:
And? You should start and stop it anyway, just wrap it with anon func